### PR TITLE
Script for Network virtualized devices

### PR DESCRIPTION
**Pull request by hathyaga**
This script will add a Network virtualized device
then does an add and remove of a Backing device
and finally removes the Network virtualized device
This will be performed as many times based on user
input

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
Job log 
[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1872563/job.log)

**Comment by narasimhan-v** [comment]
I still do not see the old script removed.

File: io/net/virt-net/Network_virtualization.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+              \"lpar_names=%s,slots=%s\"' % (server, lpar, slot_num)
+        try:
+            output = self.run_command(cmd)
+            print output
+        except CommandFailed as cf:
+            self.log.debug(str(cf))
+            self.fail("lshwres operation failed ")
+        return output
+
+    def backing_dev_add_remove(self, server, operation, lpar_id, slot_num,
```
Comment (narasimhan-v) [inline io/net/virt-net/Network_virtualization.py]: Most of the parameters are already class parameters here, can be read via `self`. So no need to pass them.

File: io/net/virt-net/Network_virtualization.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+                               add_backing_device, msg):
```
Comment (narasimhan-v) [inline io/net/virt-net/Network_virtualization.py]: can't this message be determined by the variable `operation` ?

File: io/net/virt-net/Network_virtualization.py [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+            self.log.debug(output)
+            self.fail("lshwres fails to list Network virtualized device \
+                       after add operation")
+
+    def test_network_virtualization_backing_device_add(self):
+        '''
+        Adding Backing device for Network virtualized device
+        '''
+        pre_add = self.backing_dev_count(self.slot_num)
+        self.backing_dev_add_remove(self.server, 'a', self.lpar_id,
```
Comment (narasimhan-v) [inline io/net/virt-net/Network_virtualization.py]: This should ideally be:
```
self.backing_dev_add_remove('a')
```
All other params are readable by the function also, so no need to pass them.

**Comment by hathyaga** [comment]
@narasimhan-v All suggested changes are incorporated and force pushed. Please check.

**Comment by hathyaga** [comment]
[vnic-ci-job-log.tar.gz](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1879825/vnic-ci-job-log.tar.gz)

VNIC CI job log

**Comment by hathyaga** [comment]
@narasimhan-v pep8 and pylint is taken care of. Please review.

**Comment by hathyaga** [comment]
@narasimhan-v All pep8 and pylint checks are taken care of.  Need a review.

**Comment by narasimhan-v** [comment]
@hathyaga I am not sure what you mean when you say `All pep8 and pylint checks are taken care of`
```
# pep8 Network_virtualization.py 
Network_virtualization.py:311:1: W391 blank line at end of file
```
```
# pylint Network_virtualization.py 
No config file found, using default configuration
************* Module Network_virtualization
C:311, 0: Trailing newlines (trailing-newlines)
C:  1, 0: Module name "Network_virtualization" doesn't conform to snake_case naming style (invalid-name)
R: 56, 0: Too many instance attributes (17/7) (too-many-instance-attributes)
W: 71, 8: Specify string format arguments as logging function parameters (logging-not-lazy)
```

**Comment by hathyaga** [comment]
@narasimhan-v I took care of all pep8 and pylint and force pushed network_virtualization.py

You are reviewing Network_virtualization.py which is not the right one.

**Comment by narasimhan-v** [comment]
@hathyaga Can you remove  Network_virtualization.py from the commit ?

**Comment by hathyaga** [comment]
@narasimhan-v  I removed "Network_virtualization.py" and made sure that is was no longer in the branch when I force-pushed the changes. 

This is what I have in the branch where I sent the PR
[root@ltc-zz3-lp15 virt-net]# git branch
* Network_virtualization_scripts_V2
  master
[root@ltc-zz3-lp15 virt-net]# 
[root@ltc-zz3-lp15 virt-net]# ls
Network_virtualization_driver_bind.py  Network_virtualization_driver_bind.py.data  network_virtualization.py  network_virtualization.py.data

**Comment by hathyaga** [comment]
In the interest of saving time, I've sent a new PR. 
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/854

Please merge this

**Pull request closed without merging**